scipy/stats/_fit.py

```diff
@@ -826,6 +831,21 @@ def goodness_of_fit(dist, data, *, known_params=None, fit_params=None,
     allows the user to perform the Monte Carlo trials adapted to their
     particular data.
 
+    *Algorithmic overview*
+
+    In brief, this routine executes the following steps:
+    
+      1. Fit unknown parameters to the given `data`, thereby forming the
+         "null-hypothesized" distribution and compute the statistic for them.
+      2. Draw random samples from this null-hypothesized distribution.
+      3. Re-Fit the unknown parameters again on each sample.
+      4. Calculate the statistic on each sample with the distribution specified
+         by the re-fit parameters (and the known parameters, of course).
+      5. Compare those values of the statistic (4) to the one from the `data`
+         with its corresponding null-hypothesized distribution (1).
```

Comment:
If the suggestion for steps 1 and 4 are accepted, then we would not refer to (1) and (4) here.
```suggestion
      5. Compare the value of the statistic corresponding with `data` against
         the values of the statistic corresponding with the random samples.
```

Reply:
You might define what "compare" means, while we're at it. For example, "The p-value is the proportion of samples with a statistic value greater than or equal to the statistic value from the observed data".